### PR TITLE
updates Bio.Graphics for pyton3

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request updates Bio.Graphics for python3. Minor changes only.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/Graphics/GenomeDiagram/_FeatureSet.py [hunk 1]
```diff
@@ -96,12 +96,12 @@ def set_all_features(self, attr, value):
...
         for feature in self.features.values():
-            # If the feature has the attribute, and the value should change
-            if hasattr(feature, attr):
-                if getattr(feature, attr) != value:
-                    setattr(feature, attr, value)  # set it to the passed value
+            try:
+                # If the feature has the attribute, set it to the passed value
+                setattr(feature, attr, value)
+            except AttributeError:
+                pass
```
Comment (peterjc) [inline Bio/Graphics/GenomeDiagram/_FeatureSet.py]: I can see no obvious reason for the original code here checking if the value was different, versus just setting it as you do. Should be fine.

Comment (widdowquinn) [reply Bio/Graphics/GenomeDiagram/_FeatureSet.py]: I think what I was trying to do was avoid a situation where an inappropriate attribute was set on a feature. For instance, there's not much to be gained setting `EC_number` for a `sig_peptide` feature.

I think the `try:except` approach makes sense, but if we don't check whether the feature already has the attribute, using `setattr()` will introduce the attribute where it did not exist before. That's a change of behaviour and might not be what you wanted to do.

I think:

```
 try:
    if hasattr(feature, attr):
        setattr(feature, attr, value)
```

Should keep the original behaviour.

Comment (mdehoon) [reply Bio/Graphics/GenomeDiagram/_FeatureSet.py]: `setattr` raises an `AttributeError` if the attribute doesn't exist

Comment (mdehoon) [reply Bio/Graphics/GenomeDiagram/_FeatureSet.py]: therefore we don't need to check `hasattr` first

Comment (widdowquinn) [reply Bio/Graphics/GenomeDiagram/_FeatureSet.py]: I'm not sure that's correct:

```python
Python 3.7.3 (default, Mar 27 2019, 16:54:48) 
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> class Test():
...     def __init__(self):
...             self.a = 1
... 
>>> t = Test()
>>> t.a
1
>>> t.b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Test' object has no attribute 'b'
>>> setattr(t, "b", 2)
>>> t.b
2
>>> t.__dict__
{'a': 1, 'b': 2}
```

Comment (mdehoon) [reply Bio/Graphics/GenomeDiagram/_FeatureSet.py]: I see; thanks for checking. I was trying this with a built-in class.

**Comment by mdehoon** [comment]
@widdowquinn  OK to merge?

**Comment by peterjc** [comment]
I think you've addressed @widdowquinn's review, so if there is no further comment we should merge this by the end of the week?

**Comment by widdowquinn** [comment]
Apologies for the delay. I'm happy with the merge. Many thanks @mdehoon!

**Pull request merged**